src/routes/landing/components/CitySelector.js

```diff
@@ -45,19 +53,67 @@ const developmentCompare = (a: CityModel, b: CityModel) => {
   return a.sortingName.localeCompare(b.sortingName, 'en-US')
 }
 
-class CitySelector extends React.PureComponent<PropsType> {
+const degreesToRadians = (deg: number): number => {
+  const degreesSemicircle = 180
+  return Math.PI * deg / degreesSemicircle
+}
+
+const degreeDifference = (longitude0: number, latitude0: number, longitude1: number, latitude1: number): number => {
+  return Math.acos(Math.cos(longitude0) * Math.cos(longitude1) * Math.cos(latitude0 - latitude1) +
+    Math.sin(longitude1) * Math.sin(longitude0))
+}
+
+const currentDistance = (a: CityModel, longitude: number, latitude: number) => {
+  if (a.longitude === null && a.latitude === null && a.aliases === null) {
+    return Infinity
+  }
+  const longitude0 = degreesToRadians(longitude)
+  const latitude0 = degreesToRadians(latitude)
+  const earthRadius = 6371
+  type CoordinatesType = {| longitude: number, latitude: number |}
+  // $FlowFixMe https://github.com/facebook/flow/issues/2221
+  const coordinates: Array<CoordinatesType> = Object.values(a.aliases || {})
+  coordinates.push({ longitude: a.longitude, latitude: a.latitude })
+  const distances: Array<number> = coordinates.map((coords: CoordinatesType): {| value: number |} | null => {
+    if (coords.longitude === null || coords.latitude === null) {
+      return null
+    }
+    const longitude1 = degreesToRadians(coords.longitude)
+    const latitude1 = degreesToRadians(coords.latitude)
+    return { value: degreeDifference(longitude0, latitude0, longitude1, latitude1) * earthRadius }
+  }).filter(Boolean).map(x => x.value)
+  return Math.min(...distances)
+}
+
+const compareDistance = (a: CityModel, b: CityModel, longitude: number, latitude: number) => {
+  const d0 = currentDistance(a, longitude, latitude)
+  const d1 = currentDistance(b, longitude, latitude)
+  return d0 - d1
+}
+
+type StateType = {
+  currentLongitude: null | number,
+  currentLatitude: null | number
+}
+
+class CitySelector extends React.PureComponent<PropsType, StateType> {
+  state = {
+    currentLongitude: null,
+    currentLatitude: null
+  }
```

Comment:
Since you get `currentLongitude` and `currentLatitude` via props now, you can remove them from the state and also remove the componentDidMount function.